src/piadm.sh

```diff
@@ -176,11 +176,68 @@ vcurl() {
 	fi
 }
 
-# Well-known source of SmartOS Platform Images
+# Default well-known source of SmartOS Platform Images
 DEFAULT_URL_PREFIX=https://us-east.manta.joyent.com/Joyent_Dev/public/SmartOS/
+# Default path for piadm's configuration
+PIADM_CONF=/var/piadm/piadm.conf
 
-# Can be overridden by the user's PIADM_URL_PREFIX.
-URL_PREFIX=${PIADM_URL_PREFIX:-${DEFAULT_URL_PREFIX}}
+#
+# (Re)-Configure the default URL (and potentially other things in the future)
+# using the configuration file (in $PIADM_CONF).
+#
+config_check() {
+	# Can't do standalone_only per se as it errors out, but this only
+	# applies to standalone SmartOS.
+	if [[ "$TRITON_CN" == "yes" || "$TRITON_HN" == "yes" ]]; then
+		return
+	fi
+
+	OLD_DEFAULT="$DEFAULT_URL_PREFIX"
+	if [[ -f $PIADM_CONF ]]; then
+		. $PIADM_CONF
```

Comment:
I prefer `source <file>` over `. <file>` because it's `grep`able.

Reply:
I thought `source` was csh/tcsh/zsh only.

Reply:
It works in bash too.

Reply:
This file already uses `. <PATH>` elsewhere, not changing this one.